src/components/useControlledState/__test__/useControlledState.test.tsx

```
@@ -0,0 +1,155 @@
+
+      render(<Component />);
+      expect(screen.getByTestId('state')).toHaveTextContent('lazy-default');
+    });
+
+    it('allows updating internal state', () => {
+      function Component() {
+        const [state, setState] = useControlledState(0);
+        return (
+          <div>
+            <span data-testid='state'>{state}</span>
+            <button data-testid='increment' onClick={() => setState(s => s + 1)}>
```

Biome (2.4.4)

[error] Provide an explicit type prop for the button element.

(lint/a11y/useButtonType)